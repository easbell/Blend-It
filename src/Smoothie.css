/* div .smoothie {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

div h3 {
  width: 40rem;
}
*/

img {
  width: 40rem;
  height: 40rem;
} 