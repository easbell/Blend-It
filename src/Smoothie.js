import React, { Component } from 'react';

export default class Smoothie extends Component {
  expandRecipe = () => {
    const {showRecipe} = this.props
    showRecipe(this.props)
  }
  
  render = () => {
    return (
      <div className='smoothie smoothie-in-list' onClick={this.expandRecipe}>
        <h3 className='smoothie-list-name'>{this.props.name}</h3>
        <img src={this.props.img}/>
      </div>
    )
  }
}
