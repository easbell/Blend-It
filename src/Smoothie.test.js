import React from 'react';
import Smoothie from './Smoothie.js';
import { shallow } from 'enzyme'

const mockSmoothie = {
id: 1,
name: "Babe Ruth Strawberry Pineapple Banana Smoothie",
ingredients: [
"strawberries",
"pineapple",
"banana",
"orange juice",
"yogurt",
"spinach",
"chia seeds",
"flaxseed"
],
recipe: [
"1 cup strawberries",
"1/2 cup pineapple",
"1 banana",
"2 cups orange juice",
"1/2 cup greek yogurt",
"1 cup spinach, optional",
"1 tablespoon chia or flaxseed, optional",
"ice"
],
servings: 4,
source: "https://www.modernhoney.com/6-healthy-superfood-smoothies/",
img: "images/babe-ruth.jpg"
};

const showRecipeMock = jest.fn();

describe('Smoothie', () => {
  const wrapper = shallow(
    <Smoothie
      showRecipe={showRecipeMock}
      name={mockSmoothie.name}
      img={mockSmoothie.img}
      key={mockSmoothie.id}
      id={mockSmoothie.id}
    />
  );

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change the state of isSelected and invoke showRecipe when smoothie is clicked', () => {
    wrapper.find('.smoothie').simulate('click');
    expect(showRecipeMock).toBeCalled();
  })
  

});