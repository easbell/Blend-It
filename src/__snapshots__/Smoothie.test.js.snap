// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoothie should match a snapshot with all data passed in 1`] = `
<div>
  Babe Ruth Strawberry Pineapple Banana Smoothie
</div>
`;
